kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/setupConfigurationsAndCompilations.kt

```diff
@@ -37,125 +40,194 @@ fun Project.setupConfigurationsAndCompilations(godotExtension: GodotExtension, j
-            configurations.clear()
-            configurations.add(bootstrapConfiguration)
+            val bootstrapJar = with(create("bootstrapJar", ShadowJar::class.java)) {
+                archiveBaseName.set("godot-bootstrap")
+                configurations.clear()
+                configurations.add(bootstrapConfiguration)
 
-            dependsOn(createBuildLock)
-        }
+                dependsOn(createBuildLock)
+            }
 
-        val shadowJar = with(named("shadowJar", ShadowJar::class.java).get()) {
-            archiveBaseName.set("main")
-            archiveVersion.set("")
-            archiveClassifier.set("")
-            exclude("**/module-info.class") //for android support: excludes java 9+ module info which cannot be parsed by the dx tool
+            val shadowJar = with(named("shadowJar", ShadowJar::class.java).get()) {
+                archiveBaseName.set("main")
+                archiveVersion.set("")
+                archiveClassifier.set("")
 
-            dependencies {
-                it.exclude(it.dependency("org.jetbrains.kotlin:kotlin-stdlib.*"))
-                it.exclude(it.dependency("com.utopia-rise:godot-library:.*"))
-            }
+                dependencies {
+                    it.exclude(it.dependency("org.jetbrains.kotlin:kotlin-stdlib.*"))
+                    it.exclude(it.dependency("com.utopia-rise:godot-library:.*"))
+                }
 
-            dependsOn(createBuildLock)
-        }
+                dependsOn(createBuildLock)
+            }
 
-        val checkDxToolAccessible = with(create("checkDxToolAccessible")) {
-            group = "godot-kotlin-jvm"
-            description = "Checks if the dx tool is accessible and executable. Needed for android builds only"
-
-            doLast {
-                try {
-                    val result = exec {
-                        with(it) {
-                            workingDir = projectDir
-                            isIgnoreExitValue = true
-
-                            if (DefaultNativePlatform.getCurrentOperatingSystem().isWindows) {
-                                commandLine("cmd", "/c", godotExtension.dxToolPath.get(), "--version")
-                            } else {
-                                commandLine(godotExtension.dxToolPath.get(), "--version")
+            val checkD8ToolAccessible = with(create("checkD8ToolAccessible")) {
+                group = "godot-kotlin-jvm"
+                description = "Checks if the d8 tool is accessible and executable. Needed for android builds only"
+
+                doLast {
+                    try {
+                        val d8Tool = godotExtension.d8ToolPath
+                            ?: throw IllegalArgumentException("d8 tool not set! Make sure you've either set the ANDROID_SDK_ROOT environment variable or set the d8ToolPath. For more information, visit: https://godot-kotl.in/en/stable/user-guide/exporting/#android")
```

Comment:
This one is written 3 times in that files, maybe just create your own Exception class with that text in it.

Reply:
:+1: